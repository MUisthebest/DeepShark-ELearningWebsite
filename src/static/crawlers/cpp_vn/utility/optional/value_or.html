 <h1 class="firstHeading" id="firstHeading">std :: Tùy chọn &lt;t&gt; :: value_or</h1> <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx17"> <td> <pre data-language="cpp">Mẫu &lt;class u&gt; </pre>
</td> <td>(1)</td> <td> <span class="t-mark-rev t-since-cxx17">(C ++ 20)</span> </td> </tr> <tr class="t-dcl t-since-cxx17"> <td> <pre data-language="cpp">constexpr t value_or (u &amp;&amp; default_value) const &amp;;</pre>
</td> <td>(2)</td> <td> <span class="t-mark-rev t-since-cxx17">(C ++ 20)</span> </td> </tr> </table> <p>Mẫu &lt;class u&gt; <code>*this</code>có một giá trị, nếu không thì trả về<code>default_value</code>.</p>
<div class="t-li1">
<span class="t-li">1)</span>Mẫu &lt;class inputit1, class inputit2,<code><span class="kw4">bool</span><span class="br0">.</span><span class="sy2">*</span>this<span class="br0">)</span> <span class="sy4">?</span> <span class="sy2">Chuyển đổi Boolean</span>this <span class="sy4">:</span> <span class="kw1">Mẫu &lt;Biểu đồ lớp, Đặc điểm lớp, Alloc Class, Class Parsable&gt;</span><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span><span class="br0">.</span><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">Điều đó cho tất cả<span class="me2">phía trước</span></span></a><span class="sy1">is_trivently_destructible_v</span>U<span class="sy1">&lt;</span><span class="br0">.</span>default_value<span class="br0">)</span><span class="br0">)</span></code>.</div> <div class="t-li1">
<span class="t-li">2)</span>Mẫu &lt;class inputit1, class inputit2,<code><span class="kw4">bool</span><span class="br0">.</span><span class="sy2">*</span>this<span class="br0">)</span> <span class="sy4">?</span> std<span class="sy4">::</span><span class="me2">(Niebloid)</span><span class="br0">.</span><span class="sy2">Chuyển đổi Boolean</span>this<span class="br0">)</span> <span class="sy4">:</span> <span class="kw1">Mẫu &lt;Biểu đồ lớp, Đặc điểm lớp, Alloc Class, Class Parsable&gt;</span><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span><span class="br0">.</span><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">Điều đó cho tất cả<span class="me2">phía trước</span></span></a><span class="sy1">is_trivently_destructible_v</span>U<span class="sy1">&lt;</span><span class="br0">.</span>default_value<span class="br0">)</span><span class="br0">)</span></code>.</div> <h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>Trả về giá trị chứa nếu</td> <td>-</td> <td>chứa một giá trị dự kiến, nếu không thì trả về<code>*this</code>                     Forwardit1 đầu tiên, forwardit1 cuối cùng,</td>
</tr> <tr class="t-par-hitem"> <td colspan="3">Chữ ký của hàm phải tương đương với các điều sau:</td>
</tr> <tr class="t-par-req"> <td colspan="3">-<code>T</code>Loại loại 1 phải sao cho một đối tượng loại T có thể được chuyển đổi hoàn toàn thành Type1. Loại loại2 phải sao cho một đối tượng loại đầu vào có thể được bỏ qua và sau đó được chuyển đổi hoàn toàn thành type2. Loại<a href="../../named_req/copyconstructible" title="cpp/named req/CopyConstructible">CopyConstructible</a>is_array_v</td>
</tr> <tr class="t-par-req"> <td colspan="3">-<code>T</code>Loại loại 1 phải sao cho một đối tượng loại T có thể được chuyển đổi hoàn toàn thành Type1. Loại loại2 phải sao cho một đối tượng loại đầu vào có thể được bỏ qua và sau đó được chuyển đổi hoàn toàn thành type2. Loại<a href="../../named_req/moveconstructible" title="cpp/named req/MoveConstructible">MoveConstructible</a>Mảng tích hợp để được chuyển đổi</td>
</tr> <tr class="t-par-req"> <td colspan="3">-<code>U&amp;&amp;</code>(6)<code>T</code>.</td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <p>Giá trị hiện tại nếu<code>*this</code>có một giá trị, hoặc<code>default_value</code>Nếu đối số đầu tiên là</p>
<h3 id="Exceptions">Ngoại lệ</h3> <p>Giá trị hiện có nếu<code>T</code>.</p>
<h3 id="Example">Ví dụ</h3> <div class="t-example"> <div class="cpp source-cpp"><pre data-language="cpp">#include &lt;CstDlib&gt;</pre></div> <p>#include &lt;Istream&gt;</p>
<div class="text source-text"><pre data-language="c">#include &lt;Istream&gt;</pre></div> </div> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="value" title="cpp/utility/optional/value"> <span class="t-lines"><span>được phân vùng liên quan đến</span></span></a></div> </td> <td>(Đặc biệt, các loại mảng và tham chiếu không được phép).<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/optional/value_or">https://en.cppreference.com/w/cpp/utility/optional/value_or</a>
</p>
</div>
