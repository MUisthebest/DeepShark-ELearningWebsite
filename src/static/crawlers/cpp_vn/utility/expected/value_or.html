 <h1 class="firstHeading" id="firstHeading">std :: mong đợi &lt;t, e&gt; :: value_or</h1> <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">Mẫu &lt;class u&gt; </pre>
</td> <td>(1)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">constexpr t value_or (u &amp;&amp; default_value) const &amp;;</pre>
</td> <td>(2)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> </table> <p>Mẫu &lt;class u&gt; <code>*this</code>ConstExpr t value_or (u &amp;&amp; default_value) &amp;&amp;;<code>default_value</code>.</p>
<div class="t-li1">
<span class="t-li">1)</span>Mẫu &lt;Class execPolicy,<code><span class="kw4">bool</span><span class="br0">.</span><span class="sy2">*</span>this<span class="br0">)</span> <span class="sy4">?</span> <span class="sy2">Chuyển đổi Boolean</span>this <span class="sy4">:</span> <span class="kw1">Mẫu &lt;Biểu đồ lớp, Đặc điểm lớp, Alloc Class, Class Parsable&gt;</span><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span><span class="br0">.</span><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">Điều đó cho tất cả<span class="me2">phía trước</span></span></a><span class="sy1">is_trivently_destructible_v</span>U<span class="sy1">&lt;</span><span class="br0">.</span>default_value<span class="br0">)</span><span class="br0">)</span></code>
</div> <div class="t-li1">
<span class="t-li">2)</span>Mẫu &lt;Class execPolicy,<code><span class="kw4">bool</span><span class="br0">.</span><span class="sy2">*</span>this<span class="br0">)</span> <span class="sy4">?</span> std<span class="sy4">::</span><span class="me2">(Niebloid)</span><span class="br0">.</span><span class="sy2">Chuyển đổi Boolean</span>this<span class="br0">)</span> <span class="sy4">:</span> <span class="kw1">Mẫu &lt;Biểu đồ lớp, Đặc điểm lớp, Alloc Class, Class Parsable&gt;</span><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span><span class="br0">.</span><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">Điều đó cho tất cả<span class="me2">phía trước</span></span></a><span class="sy1">is_trivently_destructible_v</span>U<span class="sy1">&lt;</span><span class="br0">.</span>default_value<span class="br0">)</span><span class="br0">)</span></code>
</div> <h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>Trả về giá trị chứa nếu</td> <td>-</td> <td>chứa một giá trị dự kiến, nếu không thì trả về<code>*this</code>Default_Value</td>
</tr> <tr class="t-par-hitem"> <td colspan="3">Chữ ký của hàm phải tương đương với các điều sau:</td>
</tr> <tr class="t-par-req"> <td colspan="3">-<code>T</code>Loại loại 1 phải sao cho một đối tượng loại T có thể được chuyển đổi hoàn toàn thành Type1. Loại loại2 phải sao cho một đối tượng loại đầu vào có thể được bỏ qua và sau đó được chuyển đổi hoàn toàn thành type2. Loại<a href="../../named_req/copyconstructible" title="cpp/named req/CopyConstructible">CopyConstructible</a>is_array_v</td>
</tr> <tr class="t-par-req"> <td colspan="3">-<code>T</code>Loại loại 1 phải sao cho một đối tượng loại T có thể được chuyển đổi hoàn toàn thành Type1. Loại loại2 phải sao cho một đối tượng loại đầu vào có thể được bỏ qua và sau đó được chuyển đổi hoàn toàn thành type2. Loại<a href="../../named_req/moveconstructible" title="cpp/named req/MoveConstructible">MoveConstructible</a>Mảng tích hợp để được chuyển đổi</td>
</tr> <tr class="t-par-req"> <td colspan="3">-<code>U&amp;&amp;</code>(6)<code>T</code> </td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <p>giá trị cần sử dụng trong trường hợp<code>*this</code>không chứa giá trị dự kiến<code>default_value</code>Nếu đối số đầu tiên là</p>
<h3 id="Exceptions">Ngoại lệ</h3> <p>Giá trị hiện có nếu<code>T</code>.</p>
<h3 id="Notes">Ghi chú</h3> <p>Nếu như<code>T</code>Chuyển đổi tiêu chuẩn được áp dụng để chuyển đổi biểu thức thành prvalue.<code>void</code>chứa một giá trị dự kiến, hoặc</p>
<h3 id="Example">Ví dụ</h3> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="value" title="cpp/utility/expected/value"> <span class="t-lines"><span>được phân vùng liên quan đến</span></span></a></div> </td> <td>kiểm tra xem đối tượng có chứa giá trị dự kiến ​​không<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/expected/value_or">https://en.cppreference.com/w/cpp/utility/expected/value_or</a>
</p>
</div>
